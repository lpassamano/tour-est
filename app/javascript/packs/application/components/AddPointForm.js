import React, { Component } from "react";
import { navigate } from "@reach/router";
import { connect } from "react-redux";
import * as pointActions from "../redux/points/actions";
import * as pointSelectors from "../redux/points/selectors";

export class AddPointForm extends Component {
  state = {
    caption: ""
  };

  handleChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  handleSubmit = async event => {
    event.preventDefault();
    const { caption } = this.state;
    const result = await this.props.onCreatePoint(this.props.tourId, {
      point: { caption: caption }
    });

    if (result.ok) {
      this.setState({ caption: "" });
      this.props.onHide();
    } else {
      console.error(result.data.error);
    }
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <p>Upload Image (WIP)</p>
        <label htmlFor="caption">Caption: </label>
        <textarea
          name="caption"
          id="caption"
          type="text"
          value={this.state.caption}
          onChange={this.handleChange}
        />
        <button type="submit" id="save">
          Save
        </button>
        <button type="button" onClick={this.props.onHide}>
          Cancel
        </button>
        <button type="button" id="save_and_add" onClick={this.handleSubmit}>
          Save and add another point
        </button>
      </form>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  tourId: ownProps.tourId,
  onHide: ownProps.onHide
});
const mapDispatchToProps = { onCreatePoint: pointActions.createPoint };

const enhance = connect(
  mapStateToProps,
  mapDispatchToProps
);

export default enhance(AddPointForm);
