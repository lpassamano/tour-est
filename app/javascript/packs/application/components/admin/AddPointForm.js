import React, { Component } from "react";
import { navigate } from "@reach/router";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import PointFormFields from "./PointFormFields";
import * as pointActions from "../../redux/points/actions";
import * as pointSelectors from "../../redux/points/selectors";

export class AddPointForm extends Component {
  handleSubmit = attributes => {
    this.props.onCreatePoint(this.props.tourId, {
      point: attributes
    });
  };

  render() {
    return (
      <PointFormFields
        onSubmit={this.handleSubmit}
        onHide={this.props.onHide}
      />
    );
  }
}

AddPointForm.propTypes = {
  tourId: PropTypes.string.isRequired,
  onHide: PropTypes.func.isRequired,
  onCreatePoint: PropTypes.func.isRequired
};

const mapDispatchToProps = { onCreatePoint: pointActions.createPoint };

const enhance = connect(
  null,
  mapDispatchToProps
);

export default enhance(AddPointForm);
