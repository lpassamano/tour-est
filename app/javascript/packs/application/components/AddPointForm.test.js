import { AddPointForm } from "./AddPointForm";
import React from "react";
import { shallow } from "enzyme";

describe("<AddPointForm />", () => {
  test("render", () => {
    const component = shallow(<AddPointForm />);
    expect(component).toMatchSnapshot();
  });

  test("handleChange - when text is typed into the fields the state is updated", () => {
    const handlePointChange = jest.fn();
    const component = shallow(
      <AddPointForm
        key={"1"}
        index={"1"}
        handlePointChange={handlePointChange}
      />
    );
    const event = { target: { name: "caption", value: "Nike of Samothrace" } };
    component.find("textarea#caption").simulate("change", event);
    expect(component.state("caption")).toEqual("Nike of Samothrace");
  });
});
