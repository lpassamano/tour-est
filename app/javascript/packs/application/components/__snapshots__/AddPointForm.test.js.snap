// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPointForm /> render 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="image"
  >
    Upload Image
  </label>
  <ImageInput
    name="image"
    onChange={[Function]}
    value={null}
  />
  <label
    htmlFor="title"
  >
    Title:
  </label>
  <input
    name="title"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <label
    htmlFor="caption"
  >
    Caption:
  </label>
  <textarea
    name="caption"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <label
    htmlFor="description"
  >
    Description:
  </label>
  <textarea
    name="description"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <label
    htmlFor="location"
  >
    Location:
  </label>
  <input
    name="location"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <label
    htmlFor="directions"
  >
    Directions:
  </label>
  <textarea
    name="directions"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <button
    type="submit"
  >
    Save
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Save and add another
  </button>
  <br />
  <button
    onClick={[MockFunction]}
    type="button"
  >
    Cancel
  </button>
</form>
`;
